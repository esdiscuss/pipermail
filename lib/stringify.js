var Transform = require('barrage').Transform
var jsesc = require('jsesc')

module.exports = stringify;
function stringify() {
  var stream = new Transform({objectMode: true})
  var first = true
  stream._transform = function (item, _, cb) {
    stream.push((first ? '' : '\n') + jsesc(item, {json: true}))
    first = false
    cb()
  }
  return stream
}