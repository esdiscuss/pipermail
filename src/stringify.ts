import {Transform} from 'barrage';

export default function stringify(): Transform<any, string> {
  let first = true;
  return new Transform({
    writableObjectMode: true,
    transform(item, _, push, cb) {
      push((first ? '' : '\n') + JSON.stringify(item));
      first = false;
      cb();
    },
  });
}
