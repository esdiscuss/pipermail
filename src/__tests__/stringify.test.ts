import stringify from '../stringify';

test('stiringify', async () => {
  const strm = stringify();
  strm.write({foo: 10});
  strm.write({bar: 42});
  strm.write({baz: 32});
  strm.end();
  expect(await strm.buffer('utf8')).toMatchSnapshot();
});
