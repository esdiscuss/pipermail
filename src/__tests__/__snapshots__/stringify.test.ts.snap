// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stiringify 1`] = `
"{\\"foo\\":10}
{\\"bar\\":42}
{\\"baz\\":32}"
`;
